stock_card_report/wizard/stock_card_report_wizard.py

```diff
@@ -0,0 +1,143 @@
+from odoo import api, fields, models
+
+
+class StockCardReportWizard(models.TransientModel):
+    _name = "stock.card.report.wizard"
+    _description = 'Stock Card Report Wizard'
+
+    date_from = fields.Date(
+        string="Start Date",
+        required=True,
+    )
+    date_to = fields.Date(
+        string="End Date",
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=True,
+    )
+    location_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Location",
+        required=True,
+    )
+    initial_stock = fields.Float(
+        string='Initial Stock',
+    )
+    line_ids = fields.One2many(
+        comodel_name='stock.card.report.wizard.line',
+        inverse_name='explosion_id',
+    )
+
+    @api.model
+    def _prepare_line(self, stock_card, type, balance, location):
+        if type == 'in':
+            return {
+                'explosion_id': self.id,
+                'product_id': stock_card.product_id.id,
+                'location_id': location,
+                'date': stock_card.date,
+                'reference': stock_card.reference,
+                'product_in': stock_card.product_uom_qty,
+                'product_out': 0.00,
+                'product_balance': balance,
+            }
+        elif type == 'out':
+            return {
+                'explosion_id': self.id,
+                'product_id': stock_card.product_id.id,
+                'location_id': location,
+                'date': stock_card.date,
+                'reference': stock_card.reference,
+                'product_in': 0.00,
+                'product_out': stock_card.product_uom_qty,
+                'product_balance': balance,
+            }
+
+    @api.multi
+    def find_stock_card(self):
+        self.ensure_one()
+        line_obj = self.env['stock.card.report.wizard.line']
+
+        def _find_initial(initial_stock_card, location):
```

Comment:
I suggest if we know it is integer, always us xxxx_id
```suggestion
        def _find_initial(initial_stock_card, location_id):
```